central/node/datastore/store/postgres/cve_store.go

```diff
@@ -0,0 +1,228 @@
+	"github.com/stackrox/rox/pkg/set"
+)
+
+// NodeCVEStore provides functionality for node CVE operations
+type NodeCVEStore interface {
+	GetCVEs(ctx context.Context, tx *postgres.Tx, cveIDs []string) (map[string]*storage.NodeCVE, error)
+	CopyFromNodeCves(ctx context.Context, tx *postgres.Tx, objs ...*storage.NodeCVE) error
+	RemoveOrphanedNodeCVEs(ctx context.Context, tx *postgres.Tx) error
+	MarkOrphanedNodeCVEs(ctx context.Context, tx *postgres.Tx) error
+}
+
+type nodeCVEStoreImpl struct {
+	cache *nodeCVECache
+}
+
+// NewNodeCVEStore creates a new NodeCVEStore instance
+func NewNodeCVEStore() NodeCVEStore {
+	return &nodeCVEStoreImpl{
+		cache: newNodeCVECache(),
+	}
+}
+
+func (s *nodeCVEStoreImpl) GetCVEs(ctx context.Context, tx *postgres.Tx, cveIDs []string) (map[string]*storage.NodeCVE, error) {
+	defer metrics.SetPostgresOperationDurationTime(time.Now(), ops.GetMany, "NodeCVEs")
+
+	// Check cache first
+	cachedCVEs, missingIDs := s.cache.GetMany(cveIDs)
+	idToCVEMap := make(map[string]*storage.NodeCVE, len(cveIDs))
+
+	// Add cached CVEs to result
+	for id, cve := range cachedCVEs {
+		idToCVEMap[id] = cve
+	}
+
+	// If all CVEs were found in cache, return early
+	if len(missingIDs) == 0 {
+		return idToCVEMap, nil
+	}
+
+	// Fetch missing CVEs from database
+	rows, err := tx.Query(ctx, "SELECT serialized FROM "+nodeCVEsTable+" WHERE id = ANY($1::text[])", missingIDs)
+	if err != nil {
+		return nil, errors.Wrap(err, "querying CVEs from database")
+	}
+	defer rows.Close()
+
+	dbCVEs := make(map[string]*storage.NodeCVE)
+	for rows.Next() {
+		var data []byte
+		if err := rows.Scan(&data); err != nil {
+			return nil, errors.Wrap(err, "scanning CVE data")
+		}
+		msg := &storage.NodeCVE{}
+		if err := msg.UnmarshalVTUnsafe(data); err != nil {
+			return nil, errors.Wrap(err, "unmarshaling CVE data")
+		}
+		dbCVEs[msg.GetId()] = msg
+		idToCVEMap[msg.GetId()] = msg
+	}
+
+	// Cache the newly fetched CVEs
+	if len(dbCVEs) > 0 {
+		s.cache.SetMany(dbCVEs)
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, errors.Wrap(err, "iterating over CVE rows")
+	}
+
+	return idToCVEMap, nil
+}
+
+func (s *nodeCVEStoreImpl) CopyFromNodeCves(ctx context.Context, tx *postgres.Tx, objs ...*storage.NodeCVE) error {
+	copyCols := []string{
+		"id",
+		"cvebaseinfo_cve",
+		"cvebaseinfo_publishedon",
+		"cvebaseinfo_createdat",
+		"operatingsystem",
+		"cvss",
+		"severity",
+		"impactscore",
+		"snoozed",
+		"snoozeexpiry",
+		"orphaned",
+		"orphanedtime",
+		"serialized",
+	}
+
+	// Process CVEs in batches using slices.Chunk
+	for batch := range slices.Chunk(objs, batchSize) {
+		// Prepare data for this batch
+		inputRows := make([][]interface{}, 0, len(batch))
+		deletes := make([]string, 0, len(batch))
+
+		for _, obj := range batch {
+			serialized, marshalErr := obj.MarshalVT()
+			if marshalErr != nil {
+				return errors.Wrapf(marshalErr, "marshaling CVE %s", obj.GetId())
+			}
+
+			inputRows = append(inputRows, []interface{}{
```

Comment:
nitpick: Could you please add a comment stating that the order must follow the columns defined in `copyCols`.

Also, moving that into a dedicated "getInputRow" function, where a global variable `copyCols` will be defined before it, would help with readability.